rootex/framework/systems/light_system.cpp

```diff
@@ -104,16 +114,18 @@ LightsInfo LightSystem::getDynamicLights()
 		return Vector3::DistanceSquared(cameraPos, aa) < Vector3::DistanceSquared(cameraPos, bb);
 	};
 
-	Vector<SpotLightComponent>& spotLightComponents = ECSFactory::GetAllSpotLightComponent();
-	sort(spotLightComponents.begin(), spotLightComponents.end(), spotLightSortingLambda);
+	ComponentArray<SpotLightComponent>& spotLightComponents = ECSFactory::GetAllSpotLightComponent();
+	// TODO: Implement sorting. Currently due to raw pointers in components it is not possible to sort without guarrateeing there
```

Comment:
I think this can still be done. We don't actually need to sort the components in memory. We only need to find what is the order of the components in terms of distance from the camera.

Maybe just create an array of pointers to these components and then sort the array based on this criteria?

Reply:
I guess we can first merge this and then make that into another PR, but it is fine either way.